src/transport.rs

```diff
@@ -118,7 +118,10 @@ pub mod http {
         }
 
         response.headers().get::<ContentLength>()
-            .ok_or_else(|| format!("expected Content-Length header, but none was found"))?;
+            .ok_or_else(|| {
+                let err: Box<Error + Send + Sync> = format!("expected Content-Length header, but none was found").into();
```

Comment:
I think it would be simpler to annotate `into`s type parameter instead, so it can stay a single expression. Try `.into<Box<...>>()`.

Reply:
method `into` is not a generic, but trait `Into` is.

Reply:
Ah, right. Can you use `Box<...>::from` instead?

Reply:
done